homeassistant/components/obihai/connectivity.py

```diff
@@ -19,16 +19,16 @@ def validate_auth(
     host: str,
     username: str,
     password: str,
-) -> bool:
+) -> tuple:
     """Test if the given setting works as expected."""
     obi = get_pyobihai(host, username, password)
 
     login = obi.check_account()
     if not login:
         LOGGER.debug("Invalid credentials")
-        return False
+        return (False, None)
 
-    return True
+    return (True, obi.get_device_serial())
```

Comment:
Is it possible to get the serial without logging in?

or MAC is also a great candidate for unique id, if it can be accessed during the manual/user process.

Reply:
I don't think Obihai will respond to any queries without authentication.  New upload using MacAddress instead.  I think it might need some cleanup...